client/src/components/Products/index.tsx

```diff
@@ -35,10 +35,16 @@ export const Products = observer(() => {
         if (categoryId === activeCategoryId) {
             setActiveCategoryId(-1)
         } else {
+            if (categoryId === 0) {
+                productStore.fetchProducts()
+                    .then(() => productStore.setStatus(StatusEnum.initial))
+            } else {
+                productStore.fetchProducts(categoryId, productStore.page)
+                    .then(() => productStore.setStatus(StatusEnum.initial))
+            }
+
             setActiveCategoryId(categoryId)
         }
```

Comment:
Some people say the more conditional you have in your code - the worse your code is.
Try to think of the way to refactor that one if else with categoryId check